File: rb/features.rb
# Abhi SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module AbhiFeatures
  def self.make_feature(name)
    case name
    when "base"
      AbhiBaseFeature.new
    when "test"
      AbhiTestFeature.new
    else
      AbhiBaseFeature.new
    end
  end
end

File: py/features.py
# Abhi SDK feature factory

from feature.base_feature import AbhiBaseFeature
from feature.test_feature import AbhiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AbhiBaseFeature(),
        "test": lambda: AbhiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
